stan/math/rev/mat/fun/log_sum_exp.hpp

```diff
@@ -14,27 +14,12 @@ namespace math {
 
 namespace internal {
 
-// these function and the following class just translate
-// log_sum_exp for std::vector for Eigen::Matrix
-
-template <int R, int C>
-inline double log_sum_exp_as_double(const Eigen::Matrix<var, R, C>& x) {
-  if (x.size() == 0) {
-    return NEGATIVE_INFTY;
-  }
-
-  const double max = x.val().maxCoeff();
-  if (!std::isfinite(max)) {
-    return max;
-  }
-  return max + std::log((x.val().array() - max).exp().sum());
-}
-
 class log_sum_exp_matrix_vari : public op_matrix_vari {
  public:
-  template <int R, int C>
-  explicit log_sum_exp_matrix_vari(const Eigen::Matrix<var, R, C>& x)
-      : op_matrix_vari(log_sum_exp_as_double(x), x) {}
+  template <typename T>
+  explicit log_sum_exp_matrix_vari(T&& x)
+      : op_matrix_vari(log_sum_exp(std::forward<T>(x).val()),
```

Comment:
This call to `std::forward` does nothing as as you only call a membor function on forwarded `x` (forwarding only makes sense if you use it when calling a function with different overloads for rvalues and lvalues or one that accepts a forwarding reference - `T&&`). If you actually forwarded one variable into two functions/classes it would not work correctly for rvalues as the first one would move content the variable and the second one would only get empty container!